util/src/main/java/tc/oc/pgm/util/StringUtils.java

```diff
@@ -140,6 +140,14 @@ public static String normalize(String text) {
             .toLowerCase(Locale.ROOT);
   }
 
+  public static String normalizeUsername(String text) {
+    return text == null
+        ? ""
+        : Normalizer.normalize(text, Normalizer.Form.NFD)
+            .replaceAll("[^A-Za-z0-9_ ]", "")
```

Comment:
Same for this

Reply:
In the past there have been legacy usernames with spaces ` ` or `.`. However I'm unable to find any at this point https://namemc.com/search?q=tempusername3 as it seems like every one I search for was changed to `tempusername#` on January 19th 2023. Not sure if there are still invalid usernames that I've missed. Not sure whether to  continue normalizing usernames given this.